ChangeLog

```diff
@@ -3,12 +3,106 @@ Changelog
 
 ## v2.4.0
 
-- Enum: uniq is modified to use structural comparison internally, instead
-    of previously used physical comparison, which was most likely a bug.
-    The old behavior is provided by the new Enum.uniqq function.
-    Enum.uniq_by was added for additional convenience.
+- ignore deprecation warnings
+  (Gabriel Scherer)
+- BatBitSet: use Bytes instead of String
+  (Gabriel Scherer)
+- rm qtest-generated file all_tests.ml upon 'make clean'
+  (Francois Berenger)
+- corrected many warnings
+  (Francois Berenger)
+- regression tests for issue #609
+  (Gabriel Scherer)
+- Makefile: create testsuite-only-{byte,native} targets
+  (Gabriel Scherer)
+- BatHashtbl: fix hash computation bug introduced by 4.01
+  (Gabriel Scherer)
+- BatText: synchronize nsplit to match BatString.nsplit
+  (Gabriel Scherer)
+- BatInt: repared quick check test for mid
+  (Francois Berenger)
+- updated NEXT_RELEASE tags
+  (Francois Berenger)
+- BatLazyList: fix remove_all_such documentation
+  (Xavier Van de Woestyne)
+- BatMap: better ocamldoc for update
```

Comment:
either skip or be more detailed "specify property X" -- as such, it does not help our users know what has changed or learn new thing about the library.